test/irmin-pack/test_gc.ml

```diff
@@ -0,0 +1,467 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+open Common
+
+let src = Logs.Src.create "tests.gc" ~doc:"Test gc"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let test_dir = "_build"
+
+let fresh_name =
+  let c = ref 0 in
+  fun () ->
+    incr c;
+    let name = Filename.concat test_dir ("test-gc" ^ string_of_int !c) in
+    name
+
+include struct
+  module S = struct
+    module Maker = Irmin_pack_unix.Maker (Conf)
+    include Maker.Make (Schema)
+  end
+
+  type t = {
+    root : string;
+    repo : S.Repo.t;
+    parents : S.Commit.t list;
+    tree : S.tree;
+  }
+
+  let config ~readonly ~fresh root =
+    Irmin_pack.config ~readonly
+      ~indexing_strategy:Irmin_pack.Indexing_strategy.minimal ~lru_size:0 ~fresh
+      root
+
+  let info = S.Info.empty
+
+  let gc ?unlink t commit =
+    let commit_key = S.Commit.key commit in
+    S.gc ?unlink t.repo commit_key
+
+  let wait_for_gc _ =
+    () (* TODO for testing purposes its convenient to have this function. *)
+
```

Comment:
It is not completely clear how we implement this function... but presumably it will be in lwt?

Reply:
Good point. We can make 2 blocking functions. One direct style and another one that returns an Lwt promise

Reply:
I changed it to Lwt, and we can add the direct style one later if needed